applications/RomApplication/tests/cpp_tests/test_lspg_rom_builder_and_solver.cpp

```diff
@@ -0,0 +1,221 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Sebastian Ares de Parga
+//
+
+// System includes
+
+/* External includes */
+
+/* Project includes */
+#include "containers/model.h"
+#include "geometries/triangle_2d_3.h"
+#include "includes/kratos_parameters.h"
+#include "testing/testing.h"
+#include "spaces/ublas_space.h"
+#include "solving_strategies/strategies/implicit_solving_strategy.h"
+#include "linear_solvers/linear_solver.h"
+#include "custom_strategies/rom_builder_and_solver.h"
+#include "custom_strategies/lspg_rom_builder_and_solver.h"
+#include "solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h"
+#include "utilities/math_utils.h"
+
+namespace Kratos {
+namespace Testing {
+namespace LSPGROMBuilderAndSolverTestingInternal {
```

Comment:
```suggestion
namespace {
```
Make it unnamed so no one can use it outside the scope of this file.